uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/ContactWorkPatternTest.php

```diff
@@ -126,4 +129,47 @@ public function testGetForDateReturnsNullIfTheEmployeeHasWorkPatternsButTheGiven
     $this->assertNull($contactWorkPattern);
   }
 
+  public function testGetWorkPatternForContact() {
+    //create a contact with valid contract and a Work pattern assigned
+    $contact = ContactFabricator::fabricate();
+    $periodStartDate = date('2016-01-01');
+
+    HRJobContractFabricator::fabricate([
+      'contact_id' => $contact['id']
+    ],
+    [
+      'period_start_date' => $periodStartDate,
+    ]);
+    $workPattern = WorkPatternFabricator::fabricateWithA40HourWorkWeek();
+    ContactWorkPatternFabricator::fabricate([
+      'contact_id' => $contact['id'],
+      'pattern_id' => $workPattern->id
+    ]);
+    $date= new DateTime('2016-01-20');
+
+    $returnedWorkPattern = ContactWorkPattern::getWorkPattern($contact['id'], $date);
+    $this->assertEquals($workPattern->id, $returnedWorkPattern->id);
+  }
```

Comment:
We need a few more tests for ContactWorkPattern::getWorkPattern()
- Test that it returns the default work pattern when the contact has no work pattern assigned to 
- Test that it returns the right work pattern when the contact has multiple work patterns assigned to
- Test that it returns the default work pattern for a contact with an assigned work pattern, when the date given to getWorkPattern is before the effective date of the work pattern